### ISSUE TITLE
Inconsistent Default Host Key

### ISSUE THREAD

**Issue by mcollier** [Azure/azure-functions-host#6892]
#### Investigative information

Please provide the following:

- Timestamp: 2020-11-12T01:59:21.4
- Function App version: 3.0.14916.0
- Function App name: https://aypnshrj6u6mgfa.azurewebsites.net
- Function name(s) (as appropriate): N/A
- Invocation ID: N/A
- Region: northcentralus

#### Repro steps

Provide the steps required to reproduce the problem:

1. Execute the ARM template located at https://github.com/mcollier/azure-functions-host-key/blob/main/azure-deploy.json.
  a. `az deployment group create -g deleteme-default-host-key -f "azure-deploy.json"`
1. Observe the Azure Functions host key as an ARM template output value, **functionHostKey**
1. Observe the Azure Functions default host key shown in the Azure portal.
1. The key shown in the portal is different than the key shown in the template output.

#### Expected behavior

The Azure Functions default host key available as part of executing the ARM template is the exact same as the default host key shown in the Azure portal.

#### Actual behavior

The default host key available after the ARM template executes is different than the Azure Functions default host key shown in the Azure portal.

#### Known workarounds

None.

#### Related information 

Provide any related information 

* Azure CLI: 2.14.2
* Azure Function Core Tools: 3.0.2996
* Azure Functions hosting plan: Consumption
* Azure Functions OS: Windows

I found a related comment at https://github.com/Azure/azure-functions-host/issues/3994#issuecomment-516741480.

**Comment by mcollier** [comment]
Is the content at https://github.com/Azure/azure-functions-host/wiki/Changes-to-Key-Management-in-Functions-V2 still accurate?

**Comment by mcollier** [comment]
Figured it out.  

TL;DR - set the FUNCTIONS_EXTENSION_VERSION to ~3 when creating the function resource (Microsoft.Web/site).

More info at https://www.michaelscollier.com/inconsistent-functions-host-key/
